src/seraphis_core/legacy_enote_types.h

```diff
@@ -47,16 +47,19 @@ namespace sp
 {
 
 ////
-// LegacyEnoteV1
+// LegacyEnoteV1 (regular & coinbase pre-RingCT, then coinbase-only post-RingCT)
 // - onetime address
 // - cleartext amount
+// - differentitiate pre/post-RingCT
 ///
 struct LegacyEnoteV1 final
 {
     /// Ko
     rct::key onetime_address;
     /// a
     rct::xmr_amount amount;
+
+    bool is_pre_rct;
 };
```

Comment:
I tried to follow your suggestion to remove `is_pre_rct` from `LegacyEnoteV1` in the most recent two commits (last one is WIP, because tests fail and I just commented out the parts where `is_pre_rct` was used and did not add all the necessary changes needed for `LegacyContextualIntermediateEnoteRecordV2` yet), but didn't find a solution for the following in `src/seraphis_mocks/mock_ledger_context.cpp`
```
const LegacyEnoteV1 *tmp_enote = enote.try_unwrap<LegacyEnoteV1>();
rct::xmr_amount amount = tmp_enote && tmp_enote->is_pre_rct ? tmp_enote->amount : 0;
```
where we need to know, if we have to use the enotes amount (if pre-rct) or a zero amount, but we only have a `LegacyEnoteVariant` not a `LegacyContextualEnoteRecord`.
And a similar "chicken-or-egg" like problem is in `src/seraphis_main/scan_balance_recovery_utils.cpp` `try_view_scan_legacy_enote_v1()`.